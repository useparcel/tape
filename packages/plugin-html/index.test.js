import { tape } from "../tape/index.ts";
import HTMLPlugin from "./index.js";

test("html plugin should not collect empty `src` attributes as assets", async () => {
  const config = {
    entry: "/index.html",
    files: {
      "/index.html": {
        content: `
          <img src="" /> 
          <img src= >
        `,
      },
    },
  };

  const results = await tape(config);
  expect(results).toMatchSnapshot();
});

test("html plugin should ignore whitepscape", async () => {
  const config = {
    entry: "/index.html",
    files: {
      "/index.html": {
        content: `
          <img src="     my-image.png     " /> 
        `,
      },
      "my-image.png": {
        content: "link to image",
      },
    },
  };

  const results = await tape(config);
  expect(results).toMatchSnapshot();
});

test.skip("html plugin should collect assets even if there aren't quotes around the attribute value", async () => {
  const config = {
    entry: "/my-image.html",
    files: {
      "/my-image.html": {
        content: `<img src=my-image.png />`,
      },
      "/should-be-slash.html": {
        content: `<img src= />`,
      },
    },
  };

  const tape = new Tape(config);
  await expect(tape()).rejects.toThrow(/my-image\.png/);

  tape.update({ entry: "/should-be-slash.html" });

  await expect(tape()).rejects.toThrow(/\//);
});

test("html plugin can ignore missing assets", async () => {
  const config = {
    entry: "/index.html",
    files: {
      "/index.html": {
        content: `<img src="my-missing-image.png" />`,
      },
    },
    plugins: [[HTMLPlugin, { ignoreMissingAssets: true }]],
  };

  const results = await tape(config);
  expect(results).toMatchSnapshot();
});
