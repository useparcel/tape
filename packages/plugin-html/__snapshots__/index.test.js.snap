// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html plugin should not collect empty \`src\` attributes as assets 1`] = `
Object {
  "diagnostics": Array [],
  "entry": "/index.html",
  "files": Object {
    "/index.html": Object {
      "content": "
          <img src=\\"\\" /> 
          <img src= >
        ",
    },
  },
}
`;

exports[`html plugin should trim asset paths 1`] = `
Object {
  "diagnostics": Array [],
  "entry": "/index.html",
  "files": Object {
    "/index.html": Object {
      "content": "
|tape-dependency          <img src=\\"     my-image.png     \\" /> 
        ",
    },
    "my-image.png": Object {
      "content": "link to image",
    },
  },
}
`;
